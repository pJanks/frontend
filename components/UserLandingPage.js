import React from "react";
import mockData from "../mockData/mockData";
import ActivityCard from "./ActivityCard";
import { View, StyleSheet, Text } from "react-native";

export default function UserLandingPage() {
  console.log(mockData);
  let userActivities = mockData.map(activity => {
    return <ActivityCard date={ activity.date } />
  })
  return (
    <View>
      { userActivities }
    </View>
  )
}

// const styles = StyleSheet.create({
//   picker: {
//     height: 40,
//     width: 250,
//     borderColor: 'gray',
//     borderWidth: 1,
//     backgroundColor: '#fff',
//   }
// });
