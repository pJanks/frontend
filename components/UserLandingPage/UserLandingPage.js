import React, { useState, useEffect } from "react";
import ActivityCard from "../ActivityCard/ActivityCard";
import { View, StyleSheet, ScrollView } from "react-native";


export default function UserLandingPage({ activities, userInfo, navigation }) {
  return (
        <ScrollView
          style={ styles.ScrollViewStyle }
          contentContainerStyle={ styles.grow }>
          {activities.map(activity => {
            return (
            <ActivityCard
            date={ activity.date }
            location={ activity.location }
            activity={ activity.activity }
            forecast={ activity.forecast }
            activityId={ activity.id }
            userId={ userInfo.id }
            navigation={navigation}
            key={ activity.id }
            />
          )})}
        </ScrollView>
  )
}

const styles = StyleSheet.create({
  ScrollViewStyle: {
    height: '78.75%',
    backgroundColor: '#b2e1f4',
  },
  grow: {
    flexGrow: 1,
    alignItems: 'center'
  }
});
