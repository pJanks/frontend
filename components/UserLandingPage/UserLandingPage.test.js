import React from 'react';
import UserLandingPage from './UserLandingPage';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const activities = [{
    "id": 1,
    "activity": "Baseball",
    "date": "2020-04-08T16:57:12Z",
    "location": "Denver, CO",
    "forecast": "Overcast",
    "forecast_img": "cloudy"
    },
    {
    "id": 2,
    "activity": "Hiking",
    "date": "2020-04-18T12:00:00Z",
    "location": "Breckenridge, CO",
    "forecast": "Sunny",
    "forecast_img": "sunny"
    }]
  const landingPage = renderer.create(<UserLandingPage activities={ activities } />).toJSON();
  expect(landingPage).toMatchSnapshot();
});
