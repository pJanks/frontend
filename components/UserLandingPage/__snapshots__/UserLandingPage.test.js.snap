// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "borderWidth": 2,
      "height": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderWidth": 1,
          "paddingBottom": 320,
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "#000",
            "borderRadius": 25,
            "borderWidth": 2,
            "height": 260,
            "margin": 30,
            "padding": 20,
            "width": "86%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25,
              "marginBottom": 10,
              "paddingHorizontal": "7%",
              "textDecorationLine": "underline",
            }
          }
        >
          Scheduled Activity:
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Scheduled on: 
          </Text>
          <Text>
            2020-04-08T16:57:12Z

          </Text>
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
             at/in: 
          </Text>
          <Text>
            Denver, CO

          </Text>
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Forecast: 
          </Text>
          <Text>
            Overcast


          </Text>
        </Text>
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#000",
                  },
                ]
              }
            >
              View Activity
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderColor": "#000",
            "borderRadius": 25,
            "borderWidth": 2,
            "height": 260,
            "margin": 30,
            "padding": 20,
            "width": "86%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 25,
              "marginBottom": 10,
              "paddingHorizontal": "7%",
              "textDecorationLine": "underline",
            }
          }
        >
          Scheduled Activity:
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Scheduled on: 
          </Text>
          <Text>
            2020-04-18T12:00:00Z

          </Text>
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
             at/in: 
          </Text>
          <Text>
            Breckenridge, CO

          </Text>
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Forecast: 
          </Text>
          <Text>
            Sunny


          </Text>
        </Text>
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#000",
                  },
                ]
              }
            >
              View Activity
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
